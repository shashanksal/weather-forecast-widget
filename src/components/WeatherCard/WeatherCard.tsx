import React from "react";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";

//Material UI Components
import { Card, CardContent, Typography, CardMedia, Grid } from "@material-ui/core";

import { WEATHER_ICONS_URL } from "../../utils/constants";

const styles = {
	card: {
		minWidth: 190
	},
	date: {
		marginBottom: 16,
		fontSize: 18
	},
	info: {
		marginBottom: 12
	},
	weatherIco: {
		width: 50,
		height: 50,
		display: "flex",
		alignItems: "center",
		justifyContent: "center",
		margin: "10px 50px"
	},
	temp: {
		fontWeight: 600,
		fontSize: 28
	}
};

const WeatherCard = (props: any) => {
	const { classes } = props;
	let icoUrl = `${WEATHER_ICONS_URL}${props.weather_state_abbr}.svg`;

	return (
		<div>
			<Card className={classes.card}>
				<CardContent>
					<Typography className={classes.date} color="textPrimary" align="center">
						{new Date(props.applicable_date).toDateString()}
					</Typography>
					<CardMedia className={classes.weatherIco} image={icoUrl} title="Weather Icon" />
					<Typography className={classes.info} color="textSecondary" align="center">
						{props.weather_state_name}
					</Typography>
					<Grid container spacing={6}>
						<Grid item md={6}>
							<Typography variant="body2" align="center">
								Max Temp (°C)
							</Typography>
							<Typography variant="body1" align="center" className={classes.temp}>
								{Math.round(props.max_temp * 10) / 10}
							</Typography>
						</Grid>
						<Grid item md={6}>
							<Typography variant="body2" align="center">
								Min Temp (°C)
							</Typography>
							<Typography variant="body1" align="center" className={classes.temp}>
								{Math.round(props.min_temp * 10) / 10}
							</Typography>
						</Grid>
					</Grid>
				</CardContent>
			</Card>
		</div>
	);
};

WeatherCard.propTypes = {
	classes: PropTypes.object.isRequired
};

export default withStyles(styles)(WeatherCard);
