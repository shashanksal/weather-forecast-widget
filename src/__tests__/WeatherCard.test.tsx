import React from "react";
import { render, fireEvent, waitForElement } from "@testing-library/react";
import WeatherCard from '../components/WeatherCard';

describe('test category', () => {
	it("WeatherCard matches snapshot" , () =>{
		const {asFragment} = render(<WeatherCard />);
		expect(asFragment()).toMatchSnapshot();
	})
})
