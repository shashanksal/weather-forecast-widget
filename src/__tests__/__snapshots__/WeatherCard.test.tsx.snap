// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test category WeatherCard matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="MuiPaper-root MuiCard-root WeatherCard-card-1 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="MuiCardContent-root"
      >
        <p
          class="MuiTypography-root WeatherCard-date-2 MuiTypography-body1 MuiTypography-colorTextPrimary MuiTypography-alignCenter"
        >
          Invalid Date
        </p>
        <div
          class="MuiCardMedia-root WeatherCard-weatherIco-4"
          style="background-image: url(https://www.metaweather.com/static/img/weather/undefined.svg);"
          title="Weather Icon"
        />
        <p
          class="MuiTypography-root WeatherCard-info-3 MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
        />
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-6"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-6"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 MuiTypography-alignCenter"
            >
              Max Temp (°C)
            </p>
            <p
              class="MuiTypography-root WeatherCard-temp-5 MuiTypography-body1 MuiTypography-alignCenter"
            >
              NaN
            </p>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-6"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 MuiTypography-alignCenter"
            >
              Min Temp (°C)
            </p>
            <p
              class="MuiTypography-root WeatherCard-temp-5 MuiTypography-body1 MuiTypography-alignCenter"
            >
              NaN
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
